Feature/css

jason-fox opened the pull request:
Adds Standard Badges and CSS as agreed by the Technical Steering Committee.

fgalan commented:
This PR collides with https://github.com/telefonicaid/fiware-orion/pull/3279. I'd suggest to merge first #3279 (which is older and larger), then rebase this one (in theory #3279 is ready to be merged, but a LGTM is yet pending).

@jason-fox what do you think?

caa06d9c commented:
Compliance test files should be fixed too, to accept new extensions.

jason-fox commented:
@fgalan - I'm happy enough to wait and rebase
@caa06d9c - I'll update `scripts/check_files_compliance.py` to allow the `favicon.ico`

fgalan commented:
PR https://github.com/telefonicaid/fiware-orion/pull/3293 was merged, so I think this PR is ready for rebase.

In addition, regarding badges, talking with @jmcanterafonseca we conclude that could be interesting to add also one for "build" status. However, I tried with

https://img.shields.io/travis/telefonicaid/fiware-orion.svg?branch=master

but it seem is not working... status shown is "unknown". Do you know why this is not working and if some adjust need to be done somewhere (e.g. repo settings) to make it work?

Thanks!

caa06d9c commented:
@fgalan we use travis to check pull requests only, there is no builds at any branches, that is why status is unknown. To enable badge we should change logic, e.g. enable travis on merge events.

fgalan commented:
@caa06d9c could be travis be enabled for master branch (without lossing the current way of working for PRs of course :) in order to have the badget? Is a .travis.yml configuration? Or something need to be adjust in repository settings?

caa06d9c commented:
Ok, I will turn it on.

caa06d9c commented on README.md:
```diff
@@ -2,11 +2,11 @@
-
-[![License badge](https://img.shields.io/badge/license-AGPL-blue.svg)](https://opensource.org/licenses/AGPL-3.0)
-[![Documentation badge](https://readthedocs.org/projects/fiware-orion/badge/?version=latest)](http://fiware-orion.readthedocs.io/en/latest/?badge=latest)
+[![FIWARE Core Context Management](https://img.shields.io/badge/FIWARE-Core-233c68.svg?logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABsAAAAVCAYAAAC33pUlAAAABHNCSVQICAgIfAhkiAAAA8NJREFUSEuVlUtIFlEUx+eO+j3Uz8wSLLJ3pBiBUljRu1WLCAKXbXpQEUFERSQF0aKVFAUVrSJalNXGgmphFEhQiZEIPQwKLbEUK7VvZrRvbr8zzjfNl4/swplz7rn/8z/33HtmRhn/MWzbXmloHVeG0a+VSmAXorXS+oehVD9+0zDN9mgk8n0sWtYnHo5tT9daH4BsM+THQC8naK02jCZ83 ...
```
Can be changed:
`[![FIWARE Core Context Management](https://nexus.lab.fiware.org/repository/raw/public/badges/chapters/core.svg)](https://www.fiware.org/developers/catalogue/)`

caa06d9c replied:
Fixed

jason-fox replied:
Fixed e42b20b

caa06d9c commented on README.md:
```diff
@@ -2,11 +2,11 @@
-[![License badge](https://img.shields.io/badge/license-AGPL-blue.svg)](https://opensource.org/licenses/AGPL-3.0)
-[![Documentation badge](https://readthedocs.org/projects/fiware-orion/badge/?version=latest)](http://fiware-orion.readthedocs.io/en/latest/?badge=latest)
+[![FIWARE Core Context Management](https://img.shields.io/badge/FIWARE-Core-233c68.svg?logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABsAAAAVCAYAAAC33pUlAAAABHNCSVQICAgIfAhkiAAAA8NJREFUSEuVlUtIFlEUx+eO+j3Uz8wSLLJ3pBiBUljRu1WLCAKXbXpQEUFERSQF0aKVFAUVrSJalNXGgmphFEhQiZEIPQwKLbEUK7VvZrRvbr8zzjfNl4/swplz7rn/8z/33HtmRhn/MWzbXmloHVeG0a+VSmAXorXS+oehVD9+0zDN9mgk8n0sWtYnHo5tT9daH4BsM+THQC8naK02jCZ83 ...
+[![License: APGL](https://img.shields.io/github/license/telefonicaid/fiware-orion.svg)](https://opensource.org/licenses/AGPL-3.0)
```
Can be changed 
`[![License badge](https://nexus.lab.fiware.org/repository/raw/public/badges/licenses/agpl.svg)](https://opensource.org/licenses/AGPL-3.0)`

jason-fox replied:
NTD - This is not a static badge, rather it reflects the type of license found in the root of the git repo and hence should be left as `img.shields.io`

caa06d9c replied:
How? 'https://img.shields.io/badge/license-AGPL-blue.svg' points to a static file, that does not belong to projects, you can compare it with 'https://img.shields.io/travis/telefonicaid/fiware-orion.svg'. So we can change it to Nexus too.

jason-fox replied:
Your right - the current badge is incorrect,The License Badge **Should** read the license
 Fixed 37c2e2ed5

caa06d9c commented on README.md:
```diff
@@ -2,11 +2,11 @@
+[![FIWARE Core Context Management](https://img.shields.io/badge/FIWARE-Core-233c68.svg?logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABsAAAAVCAYAAAC33pUlAAAABHNCSVQICAgIfAhkiAAAA8NJREFUSEuVlUtIFlEUx+eO+j3Uz8wSLLJ3pBiBUljRu1WLCAKXbXpQEUFERSQF0aKVFAUVrSJalNXGgmphFEhQiZEIPQwKLbEUK7VvZrRvbr8zzjfNl4/swplz7rn/8z/33HtmRhn/MWzbXmloHVeG0a+VSmAXorXS+oehVD9+0zDN9mgk8n0sWtYnHo5tT9daH4BsM+THQC8naK02jCZ83 ...
+[![License: APGL](https://img.shields.io/github/license/telefonicaid/fiware-orion.svg)](https://opensource.org/licenses/AGPL-3.0)
+[![Documentation badge](https://img.shields.io/readthedocs/fiware-orion.svg)](https://fiware-orion.rtfd.io)
 [![Docker badge](https://img.shields.io/docker/pulls/fiware/orion.svg)](https://hub.docker.com/r/fiware/orion/)
```
Can be changed:
`[![SOF support badge](https://nexus.lab.fiware.org/repository/raw/public/badges/stackoverflow/orion.svg)](http://stackoverflow.com/questions/tagged/fiware-orion)`

jason-fox replied:
Fixed e43f0a399

caa06d9c replied:
Nexus stores rpm packages of CB itself, so badges are a small problem. @flopezag is working on it.

jason-fox commented:
RE - travis & NGSI v2 badges - fixed e43f0a399

jason-fox commented on README.md:
```diff
@@ -2,11 +2,11 @@
      release.sh needs to be changed also -->
-
-[![License badge](https://img.shields.io/badge/license-AGPL-blue.svg)](https://opensource.org/licenses/AGPL-3.0)
-[![Documentation badge](https://readthedocs.org/projects/fiware-orion/badge/?version=latest)](http://fiware-orion.readthedocs.io/en/latest/?badge=latest)
```
* The `img.shields.io` badge will read `version=latest` by default
* `https://fiware-orion.rtfd.io` is just shorthand is a fixed version`https://fiware-orion.rtfd.io/en/1.14.0/`

fgalan replied:
So, release.sh should be adapted to fix the sed statement adjusting that literals based in the new patterns, isn't it?

jason-fox replied:
URLs updated on README and release script Fixed a1ce4111

jason-fox commented:
@fgalan @caa06d9c - I've rebased and fixed the comments based on the previous review. Could you check again and add details of any further changes necessary.

It will be much easier to confirm further changes against the rebased branch in the right location.

fgalan commented on doc/manuals.jp/index.md:
```diff
@@ -1,5 +1,8 @@
 # Orion Context Broker へようこそ
 
+[![FIWARE Core Context Management](https://img.shields.io/badge/FIWARE-Core-233c68.svg?logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABsAAAAVCAYAAAC33pUlAAAABHNCSVQICAgIfAhkiAAAA8NJREFUSEuVlUtIFlEUx+eO+j3Uz8wSLLJ3pBiBUljRu1WLCAKXbXpQEUFERSQF0aKVFAUVrSJalNXGgmphFEhQiZEIPQwKLbEUK7VvZrRvbr8zzjfNl4/swplz7rn/8z/33HtmRhn/MWzbXmloHVeG0a+VSmAXorXS+oehVD9+0zDN9mgk8n0sWtYnHo5tT9daH4BsM+THQC8naK02jCZ83 ...
+[![](https://img.shields.io/badge/tag-fiware--orion-orange.svg?logo=stackoverflow)](http://stackoverflow.com/questions/tagged/fiware-orion)
```
Should be changed? In the main README.md the following one is being used:

```
[![SOF support badge](https://nexus.lab.fiware.org/repository/raw/public/badges/stackoverflow/orion.svg)](http://stackoverflow.com/questions/tagged/fiware-orion)
```

jason-fox replied:
Badge Fixed 30574d4

fgalan commented on doc/manuals.jp/index.md:
```diff
@@ -1,5 +1,8 @@
 # Orion Context Broker へようこそ
 
+[![FIWARE Core Context Management](https://img.shields.io/badge/FIWARE-Core-233c68.svg?logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABsAAAAVCAYAAAC33pUlAAAABHNCSVQICAgIfAhkiAAAA8NJREFUSEuVlUtIFlEUx+eO+j3Uz8wSLLJ3pBiBUljRu1WLCAKXbXpQEUFERSQF0aKVFAUVrSJalNXGgmphFEhQiZEIPQwKLbEUK7VvZrRvbr8zzjfNl4/swplz7rn/8z/33HtmRhn/MWzbXmloHVeG0a+VSmAXorXS+oehVD9+0zDN9mgk8n0sWtYnHo5tT9daH4BsM+THQC8naK02jCZ83 ...
```
Should be changed? In the main README.md the following one is being used:

```
[![FIWARE Core Context Management](https://nexus.lab.fiware.org/repository/raw/public/badges/chapters/core.svg)](https://www.fiware.org/developers/catalogue/)
```

jason-fox replied:
Badge Fixed 30574d4

jason-fox commented on doc/manuals/index.md:
```diff
@@ -1,8 +1,12 @@
 # Welcome to Orion Context Broker.
 
+[![FIWARE Core Context Management](https://img.shields.io/badge/FIWARE-Core-233c68.svg?logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABsAAAAVCAYAAAC33pUlAAAABHNCSVQICAgIfAhkiAAAA8NJREFUSEuVlUtIFlEUx+eO+j3Uz8wSLLJ3pBiBUljRu1WLCAKXbXpQEUFERSQF0aKVFAUVrSJalNXGgmphFEhQiZEIPQwKLbEUK7VvZrRvbr8zzjfNl4/swplz7rn/8z/33HtmRhn/MWzbXmloHVeG0a+VSmAXorXS+oehVD9+0zDN9mgk8n0sWtYnHo5tT9daH4BsM+THQC8naK02jCZ83 ...
```
Badges Fixed 30574d4

fgalan commented on scripts/build/release.sh:
```diff
@@ -168,19 +168,19 @@ touch CHANGES_NEXT_RELEASE
 if [ "$BROKER_RELEASE" != "dev" ]
 then
-  sed "s/(https:\/\/readthedocs.org\/projects\/fiware-orion\/badge\/?version=latest)](http:\/\/fiware-orion.readthedocs.io\/en\/latest\/?badge=latest)/(https:\/\/readthedocs.org\/projects\/fiware-orion\/badge\/?version=$NEW_VERSION)](http:\/\/fiware-orion.readthedocs.io\/en\/$NEW_VERSION\/?badge=$NEW_VERSION)/" README.md > /tmp/README.md
+  sed "s/(https:\/\/img.shields.io\/readthedocs\/fiware-orion.svg)](https:\/\/fiware-orion.rtfd.io)/(https:\/\/img.shields.io\/readthedocs\/fiware-orion.$NEW_VERSION.svg)](https:\/\/fiware-orion.rtfd.io\/en\/$NEW_VERSION\/)/" README.md > /tmp/README.md
```
Let's test the new sed command in a release scenario (i.e. `[ "$BROKER_RELEASE" != "dev" ]` evaluates to true). Let current version be 1.14.0-next and new version 1.15.0.

Input line (as it is in README.md):
```
[![Documentation badge](https://img.shields.io/readthedocs/fiware-orion.svg)](https://fiware-orion.rtfd.io) 
```
Test:

```
$ export NEW_VERSION=1.15.0
$ export currentVersion=1.14.0-next
$ echo '[![Documentation badge](https://img.shields.io/readthedocs/fiware-orion.svg)](https://fiware-orion.rtfd.io)' | sed "s/(https:\/\/img.shields.io\/readthedocs\/fiware-orion.svg)](https:\/\/fiware-orion.rtfd.io)/(https:\/\/img.shields.io\/readthedocs\/fiware-orion.$NEW_VERSION.svg)](https:\/\/fiware-orion.rtfd.io\/en\/$NEW_VERSION\/)/"
```

Result:

```
[![Documentation badge](https://img.shields.io/readthedocs/fiware-orion.1.15.0.svg)](https://fiware-orion.rtfd.io/en/1.15.0/)
```

* https://img.shields.io/readthedocs/fiware-orion.1.15.0.svg, doesn't work (it shows "docs | inaccesible")
* https://fiware-orion.rtfd.io/en/1.15.0/ does work

jason-fox replied:
Fixed 18d31bc6 - it should be a`/` separator, not a `.` between the repo name and the version.

fgalan replied:
I do confirm that this couple of seds work fine:

Relasing:

```
$ export NEW_VERSION=1.15.0
$ export currentVersion=1.14.0-next
$ echo '[![Documentation badge](https://img.shields.io/readthedocs/fiware-orion.svg)](https://fiware-orion.rtfd.io)' |sed "s/(https:\/\/img.shields.io\/readthedocs\/fiware-orion.svg)](https:\/\/fiware-orion.rtfd.io)/(https:\/\/img.shields.io\/readthedocs\/fiware-orion\/$NEW_VERSION.svg)](https:\/\/fiware-orion.rtfd.io\/en\/$NEW_VERSION\/)/"
[![Documentation badge](https://img.shields.io/readthedocs/fiware-orion/1.15.0.svg)](https://fiware-orion.rtfd.io/en/1.15.0/)
```

Opening new development cycle:

```
$ export currentVersion=1.15.0
$ echo '[![Documentation badge](https://img.shields.io/readthedocs/fiware-orion/1.15.0.svg)](https://fiware-orion.rtfd.io/en/1.15.0/)' | sed "s/(https:\/\/img.shields.io\/readthedocs\/fiware-orion\/$currentVersion.svg)](https:\/\/fiware-orion.rtfd.io\/en\/$currentVersion\/)/(https:\/\/img.shields.io\/readthedocs\/fiware-orion.svg)](https:\/\/fiware-orion.rtfd.io)/"
[![Documentation badge](https://img.shields.io/readthedocs/fiware-orion.svg)](https://fiware-orion.rtfd.io)
```

fgalan commented on scripts/build/release.sh:
```diff
@@ -168,19 +168,19 @@ touch CHANGES_NEXT_RELEASE
 
 # Adjust Readthedocs documentation link for GET /version response. Note that the procedure is not symmetric 
 # (like in version.h), as dev release sets 'master' and not 'X.Y.Z-next"
 if [ "$BROKER_RELEASE" != "dev" ]
```
Looking to this I have realized (sorry for not mentioning before!) that the string literal in https://github.com/telefonicaid/fiware-orion/blob/master/src/lib/common/defaultValues.h#L44 should be also changed. Otherwise the seds are not going to work...

The change in defaultValues.h will probably needs some adjust in .test (the ones using GET /version). Travis will tell which ones are failing after the change.

fgalan commented on scripts/build/release.sh:
```diff
@@ -168,19 +168,19 @@ touch CHANGES_NEXT_RELEASE
 else
-  sed "s/https:\/\/fiware-orion.readthedocs.org\/en\/$currentVersion\//https:\/\/fiware-orion.readthedocs.org\/en\/master\//" src/lib/common/defaultValues.h > /tmp/defaultValues.h
+  sed "s/https:\/\/fiware-orion.rtfd.io\/en\/$currentVersion\//https:\/\/fiware-orion.rtfd.io\//" src/lib/common/defaultValues.h > /tmp/defaultValues.h
 fi
```
Se second pair of sed also seems to work fine:

At release time:

```
$ export NEW_VERSION=1.15.0
$ export currentVersion=1.14.0-next
$ echo '"doc" : "https://fiware-orion.rtfd.io/"' | sed "s/https:\/\/fiware-orion.rtfd.io\//https:\/\/fiware-orion.rtfd.io\/en\/$NEW_VERSION\//"
"doc" : "https://fiware-orion.rtfd.io/en/1.15.0/"
```

At the time of opening a new version devel cycle:

```
$ export currentVersion=1.15.0
$ echo '"doc" : "https://fiware-orion.rtfd.io/en/1.15.0/"' | sed "s/https:\/\/fiware-orion.rtfd.io\/en\/$currentVersion\//https:\/\/fiware-orion.rtfd.io\//"
"doc" : "https://fiware-orion.rtfd.io/"
```

NTC (just informative)

fisuda commented:
@jason-fox Thank you for updating the Japanese documents.
LGTM

caa06d9c commented:
LGTM. Thanks!
